nats.go

```diff
@@ -1164,20 +1194,26 @@ func (nc *Conn) createConn() (err error) {
 
 // makeTLSConn will wrap an existing Conn using TLS
 func (nc *Conn) makeTLSConn() {
-	// Allow the user to configure their own tls.Config structure, otherwise
-	// default to InsecureSkipVerify.
+	// Allow the user to configure their own tls.Config structure,
+	// otherwise default to InsecureSkipVerify.
 	// TODO(dlc) - We should make the more secure version the default.
+	var tlsCopy *tls.Config
 	if nc.Opts.TLSConfig != nil {
-		tlsCopy := util.CloneTLSConfig(nc.Opts.TLSConfig)
-		// If its blank we will override it with the current host
-		if tlsCopy.ServerName == _EMPTY_ {
+		tlsCopy = util.CloneTLSConfig(nc.Opts.TLSConfig)
+	} else {
+		tlsCopy = &tls.Config{InsecureSkipVerify: true}
+	}
+	// If its blank we will override it with the current host
+	if tlsCopy.ServerName == _EMPTY_ {
+		s := nc.srvPool[0]
```

Comment:
0 for reconnects correspond indeed to the current url in the pool, but not in the case of initial connect. I wonder if instead of having nc.url for the current url we should have stored nc.srv for the current *srv, from which we can then access url or tlsName.

Reply:
Might be a good idea. Let me look into that.

Reply:
Confused. You did save current and replaced everywhere but still are using srvPool[0] here?

Reply:
uug, too early in the morning. Will fix.